lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -105,7 +106,49 @@ define((require, exports, module) ->
                 result.mismatch = false
         return result
 
+
+    popoverHandler = popoverHandler ? {
```

Comment:
copy'n'paste is nice method, but it requires some thinking about what is really needed and what can or should be skipped. Here you copied code which uses external implementation of popover-managing code (`popoverHandler`) if it is supplied and its own implementation if external one is missing. However, there is no place in your code which could supply this external implementation -- you do not pass any instance of `popoverHandler` to this code. So it is equivalent to just 

```
popoverHandler = 
   options:
   ....
```
I think you need to provide API for plugging UI for this message about the range. It could look like a function with two arguments: left and right ends of the range. Pass the function as an rgument to `highlightBrackets`. If function is undefined then use your own function with this popover stuff.